modules/public/logic.py

```diff
@@ -0,0 +1,258 @@
+import bleach
+from math import floor
+import pycountry
+import re
+from urllib.parse import parse_qs, urlparse
+
+
+def calculate_colors(countries, max_users):
+    """Calculate the displayed colors for a list of country depending on the
+    maximum of users
+
+    :param countries: List of countries
+    :param max_users: Maximum of users
+
+    :returns: The list of countries with a caculated color for each
+    """
+    for country_code in countries:
+        countries[country_code]['color_rgb'] = [
+            calculate_color(
+                countries[country_code]['percentage_of_users'],
+                max_users,
+                8,
+                229
+            ),
+            calculate_color(
+                countries[country_code]['percentage_of_users'],
+                max_users,
+                64,
+                245
+            ),
+            calculate_color(
+                countries[country_code]['percentage_of_users'],
+                max_users,
+                129,
+                223
+            )
+        ]
+
+
+def calculate_color(thisCountry, maxCountry, maxColor, minColor):
+    """Calculate displayed color for a given country
+
+    :param thisCountry: The country
+    :param maxCountry: The number of users of the country with the most users
+    :param maxColor: The maximum color to reach
+    :param minColor: The minimum color to reach
+
+    :returns: The calculated color for the country
+    """
+    countryFactor = float(thisCountry)/maxCountry
+    colorRange = maxColor - minColor
+    return int(colorRange*countryFactor+minColor)
+
+
+def transform_metrics(geodata):
+    """Transform metrics to get more displayable metrics
```

Comment:
From reading this docstring, I still don't understand what this function does. Could you elaborate?